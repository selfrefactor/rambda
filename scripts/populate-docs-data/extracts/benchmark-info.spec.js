import { benchmarkInfo } from './benchmark-info.js'

test('happy', async () => {
  expect(await benchmarkInfo()).toMatchSnapshot()
})
