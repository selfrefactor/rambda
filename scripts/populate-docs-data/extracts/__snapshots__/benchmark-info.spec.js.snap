// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Object {
  "add": Object {
    "benchmarkContent": "const add = [
  {
    label : 'Rambda',
    fn    : () => {
      R.add(1, 1)
    },
  },
  {
    label : 'Ramda',
    fn    : () => {
      Ramda.add(1, 1)
    },
  },
  {
    label : 'Lodash',
    fn    : () => {
      _.add(1, 1)
    },
  },
]",
    "methodSummary": "Rambda is fastest. Ramda is 28.16% slower and Lodash is 76.17% slower",
  },
  "adjust": Object {
    "benchmarkContent": "const list = [ 0, 1, 2 ]
const fn = x => x + 1
const index = 1

const adjust = [
  {
    label : 'Rambda',
    fn    : () => {
      R.adjust(index, fn, list)
      R.adjust(index, fn)(list)
    },
  },
  {
    label : 'Ramda',
    fn    : () => {
      Ramda.adjust(index, fn, list)
      Ramda.adjust(index, fn)(list)
    },
  },
]",
    "methodSummary": "Rambda is faster than Ramda with 2.8%",
  },
  "all": Object {
    "benchmarkContent": "const input = [ 1, 2, 3, 4 ]

const all = [
  {
    label : 'Rambda',
    fn    : () => {
      const fn = x => x > 2

      R.all(fn, input)
      R.all(fn)(input)
    },
  },
  {
    label : 'Ramda',
    fn    : () => {
      const fn = x => x > 2

      Ramda.all(fn, input)
      Ramda.all(fn)(input)
    },
  },
]",
    "methodSummary": "Rambda is faster than Ramda with 89.64%",
  },
}
`;
